cpp/include/cuspatial/types.hpp

```diff
@@ -16,18 +16,39 @@
 
 #pragma once
 
+#ifdef __CUDACC__
+#define CUSPATIAL_HOST_DEVICE __host__ __device__
+#else
+#define CUSPATIAL_HOST_DEVICE
+#endif
+
+#include <cstdint>
+
 namespace cuspatial {
 
+/**
+ * @brief A 2D location: latitude and longitude, altitude
+ *
+ * @tparam T the base type for the coordinates
+ */
+template <typename T>
+struct location_2d {
+  using value_type = T;
+  value_type longitude;
+  value_type latitude;
+};
```

Comment:
Take care here. This has an insidious performance issue where the alignment of `location_2d` is 4 instead of 8. So the load/stores are going to be less efficient.

e.g., https://godbolt.org/z/Exx9WfaWj

```suggestion
template <typename T>
struct alignas(2*sizeof(T)) location_2d {
  using value_type = T;
  value_type longitude;
  value_type latitude;
};
```